packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -200,4 +245,54 @@ export class HttpApi extends Resource implements IHttpApi {
       integration: options.integration,
     }));
   }
+
+  /**
+   * Add a custom domain for this API and create the API mapping to a stage.
+   */
+  public addDomainName(options: AddDomainNameOptions): DomainName {
```

Comment:
This is the wrong modeling here. Domain Names are completely separate from HttpApi. This API implies that that a domain name is a sub-type of HttpApi which is not true. 

Drop this method entirely. Customers should create a domain name using `new DomainName()` constructor.